translator/cmdutil/userutil_linux.go

```diff
@@ -6,29 +6,108 @@
 package cmdutil
 
 import (
+	"bufio"
+	"bytes"
 	"fmt"
+	"golang.org/x/sys/unix"
+	"io"
 	"log"
 	"os"
-	"syscall"
+	"os/user"
+	"strconv"
+	"strings"
+)
 
-	"golang.org/x/sys/unix"
+type ExecUser struct {
+	Uid  int
+	Gid  int
+	Home string
+	Gids []int
+}
 
-	"github.com/opencontainers/runc/libcontainer/system"
-	"github.com/opencontainers/runc/libcontainer/user"
-)
+func getGroupIds(u *user.User) ([]int, error) {
+	file, err := os.Open("/etc/group")
+	if err != nil {
+		log.Printf("E! Failed to open group file: %v", err)
+		return nil, err
+	}
+	defer file.Close()
+	reader := bufio.NewReader(file)
+	groupIds := []int{}
+	for {
+		var wholeLine []byte
+		for {
+			line, isPrefix, err := reader.ReadLine()
```

Comment:
My guess is we don't want to do anything super bloated, but I noticed this from the `bufio` docs:
> ReadLine is a low-level line-reading primitive. Most callers should use ReadBytes('\n') or ReadString('\n') instead or use a Scanner.

Should we change the call to something else? It does say _most_, and I'm not really opposed to how it's written currently. Just figured I'd bring it up while I was looking at the docs.

Reply:
I saw that, but mainly based it on the golang implementation: https://golang.org/src/os/user/lookup_unix.go